### PR TITLE
Logs of multiple containers in a workload

### PR DESCRIPTION
**Pull request by tbeerbower**
# Description

Current containerized and helidon workload's fluentd sidecar implementation assumes there is only one application-container in the generated k8s Deployment. OAM ContainerizedWorkload and Deployment allow multiple containers.

Refactor helidon logging: 
- add container name to ElasticSearch index
- add `oam.container.name` to logged fields 
- create source, filter and match in fluentd config for each app container 
- single fluent sidecar container and config map works for all app containers

### Tests
- unit tests for helidon ContainerizedWorkloads and VerrazzanoHelidonWorkload
- update integration test
- add helidon logging acceptance tests

Fixes VZ-2017

# Checklist 

As the author of this PR, I have:

- [x] Checked that I included or updated copyright and license notices in all files that I altered
- [x] Added or updated unit tests for any new functions I added
- [x] Added or updated integration tests if appropriate
- [x] Added or updated acceptance tests if appropriate

Code reviewer, please confirm this PR:

- [ ] Addressed the requirement and meets the acceptance criteria
- [ ] Does not introduce unrelated or spurious changes
- [ ] Does not introduce any unapproved dependency
- [ ] Makes sense and it easy to understand, and/or difficult areas of code are clearly documented so that they can be understood

### CODE REVIEW

File: application-operator/apis/oam/v1alpha1/loggingscope_types.go [hunk 1]
```diff
@@ -24,10 +24,18 @@ type LoggingScopeSpec struct {
 	// Name of secret with ElasticSearch credentials
 	SecretName string `json:"secretName"`
 
+	Paths map[string]LoggingScopePath `json:"paths"`
```
Comment (fredtibbitts) [inline application-operator/apis/oam/v1alpha1/loggingscope_types.go:27]: Just a heads up, I have pending changes that make all of the logging scope fields optional (except for workloadRefs), if this new field will have a default then you should add `omitempty`.

Comment (tbeerbower) [reply application-operator/apis/oam/v1alpha1/loggingscope_types.go]: I realized that I didn't need this to resolve the issue.  I think that adding container specific fields to a logging scope needs a longer look but should not be part of this PR.

File: application-operator/controllers/loggingscope/helidon.go [hunk 2]
```diff
@@ -421,19 +424,25 @@ func replicateVmiSecret(vmiSec *kcore.Secret, namespace, name string) *kcore.Sec
 	return sec
 }
 
-func (h *HelidonHandler) ensureFluentdConfigMap(ctx context.Context, namespace, workloadName string) error {
+func (h *HelidonHandler) ensureFluentdConfigMap(ctx context.Context, namespace, workloadName string, appContainersNames []string) (err error) {
+
+	helidonFluentdConfiguration := HelidonFluentdConfiguration
+	// add the container specific configuration
+	for _, containerName := range appContainersNames {
+		helidonFluentdConfiguration += fmt.Sprintf(HelidonFluentdContainerConfigurationFormat, containerName, containerName, workloadName, containerName, workloadName, containerName, containerName, workloadName, containerName, containerName)
```
Comment (shihchang) [inline application-operator/controllers/loggingscope/helidon.go:432]: Would you try https://golang.org/pkg/text/template/ to make the template and this line more readable?

**Review by shihchang — APPROVED** [review]

File: application-operator/controllers/loggingscope/helidon.go (hunk 2 above)
Comment (tbeerbower) [reply application-operator/controllers/loggingscope/helidon.go]: Thanks for the suggestion.  Done.

**Pull request merged**